numba/core/typed_passes.py

```diff
@@ -91,14 +92,16 @@ def run_pass(self, state):
         with fallback_context(state, 'Function "%s" failed type inference'
                               % (state.func_id.func_name,)):
             # Type inference
-            typemap, return_type, calltypes = type_inference_stage(
+            typemap, return_type, calltypes, errs = type_inference_stage(
                 state.typingctx,
                 state.func_ir,
                 state.args,
                 state.return_type,
                 state.locals,
                 raise_errors=self._raise_errors)
             state.typemap = typemap
+            # save errors in case of partial typing
+            state.typing_errors = errs
```

Comment:
Instead of saving only the last typing error, should it save all errors in a dictionary that is keyed by the `pass_id`?

Reply:
We access typing errors within the same pass (after calling `run_pass` of parent), so having `pass_id` is not needed. I think the added complexity is not worth it, especially because `typemap` and `calltypes` are unique as well.

Reply:
ok, i agree the extra complexity is not worth it for now.